### PR TITLE
Pass `--set-gtid-purged=OFF` to app binding for stash

### PR DESCRIPTION
**Pull request by heheh13**
… restore

Signed-off-by: Mehedi Hasan <mehedi.hasan@appscode.com>

### CODE REVIEW

File: pkg/controller/appbinding.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ func (c *Controller) ensureAppBinding(db *api.MySQL) (kutil.VerbType, error) {
 		in.Spec.ClientConfig.InsecureSkipTLSVerify = false
 
 		// If group replication is used, then we have to pass "--set-gtid-purged=OFF" in the backup task parameter.
-		if db.UsesGroupReplication() || db.IsInnoDBCluster() {
+		if db.UsesGroupReplication() || db.IsInnoDBCluster() || db.IsSemiSync() || db.IsReadReplica() {
```
Comment (hossainemruz) [inline pkg/controller/appbinding.go:91]: Can we reverse the logic like this?

```go
if !db.IsStandalone(){
}
```

**Review by hossainemruz — CHANGES_REQUESTED** [review]

File: pkg/controller/appbinding.go (hunk 1 above)
Comment (heheh13) [reply pkg/controller/appbinding.go:91]: yes, thanks that will be better.
I planning to use something like isGTIDBased replication

File: pkg/controller/appbinding.go [hunk 2]
```diff
@@ -38,6 +38,13 @@ import (
 	ofst "kmodules.xyz/offshoot-api/api/v1"
 )
 
+func IsGTIDBasedReplication(db *api.MySQL) bool {
```
Comment (hossainemruz) [inline pkg/controller/appbinding.go:41]: Why these functions are public? Are they being used from other package?

**Review by hossainemruz — CHANGES_REQUESTED** [review]

File: pkg/controller/appbinding.go (hunk 2 above)
Comment (heheh13) [reply pkg/controller/appbinding.go:41]: Planning to move this in apimachinary helper, there are some other senerioes where I had to check these conditions at the same time

Comment (hossainemruz) [reply pkg/controller/appbinding.go:41]: ok

**Review by hossainemruz — CHANGES_REQUESTED** [review]

**Review by hossainemruz — APPROVED** [review]

**Review by spectro30 — APPROVED** [review]

**Pull request merged**